src/payment/strategies/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -53,4 +55,14 @@ export default class PaypalCommercePaymentStrategy implements PaymentStrategy {
     deinitialize(): Promise<InternalCheckoutSelectors> {
         return Promise.resolve(this._store.getState());
     }
+
+    private async _getOrderId(): Promise<string> {
+        const state = this._store.getState();
+        const cart = state.cart.getCartOrThrow();
+        const { approveUrl, orderId } = await this._paypalCommerceRequestSender.setupPayment(cart.id);
```

Comment:
I think I asked this question before but I'm not sure if I understand this correctly. I assume after this `setupPayment` call, an order ID is set in the session. Imagine the shopper cancels the PayPal modal (triggered by the next line `paymentPayPal`), and they refresh the page in order to try again. This time, because `initializationData.orderId` is defined, `getOrderId` won't be called again and therefore won't trigger the `paymentPayPal` modal?

Reply:
Order ID isn't set in the session. When shopper refresh the page `orderId` is undefined and this functional repeated

Reply:
@davidchin it is a good question, currently, payment ID passed to the Payment checkout in `onApprove` action https://github.com/bigcommerce/checkout-sdk-js/blob/2cc3cb83ce88a2aa8da0231c9425b124b892e29c/src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-button-strategy.ts#L68, only after customer approved payment session on PayPal side. 
However, we are going to revisit this workflow soon enough in the scope of ticket https://jira.bigcommerce.com/browse/PAYPAL-307, and it will work in the way you described (payment ID will be saved into the session during `setupPayment` call)

Reply:
Right I see. Yes, so the thing that I'm confused with is what happens when `POST /api/storefront/payment/paypalcommerce` (`setupPayment`) is called. From I can I see, the server gets an order ID from PayPal, and then sets it in the session before returning to the client. If that's the case, then we'll hit the problem I described above. But perhaps I'm looking at the wrong part of the code. 🤷‍♂️Anyway, I'll message you this afternoon and we can clear this up :)

Reply:
@davidchin I have tried the case you mentioned above, and you are absolutely right, good catch! 
I have removed part of the code that saves ID into the session, this part of early refactoring that I mentioned before can cause issues, because of it was added before Checkout flow and after Cart flow, it passed without any issues. The flow on the FE for this feature works well and should not cause any issues with a session, however, once we decide to continue refactoring of the way we treat order id it will require changes on FE.